### PR TITLE
Support conditional questions in securedrop-admin sdconfig.

### LINKED ISSUE #3391: Support context-dependent behavior in securedrop-admin sdconfig
## Description 

Currently, the default behavior in `securedrop-admin sdconfig` is that each question will be displayed, even if the question is not relevant for the SecureDrop instance. An enhancement would be to support asking optional questions only when necessary. For example, only an instance that answers "yes" to wanting to configure HTTPS on their source interface needs to answer the three questions added in #3366. This functionality can then replace the approach taken in #3340, which worked around this limitation to implement this behavior for the journalist notifications. 

## User Stories

As a SecureDrop administrator, I want to only answer necessary questions such that it is faster to install SecureDrop.

### PR DESCRIPTION
**Pull request by antsar**
## Status

Ready for review

## Description of Changes

Add support for conditional prompts in the `securedrop-admin sdconfig`. This adds a new field to each question. This field should be a function. If it returns `True`, the question will be asked.

This fixes #3391.

### Example usage

Note the last line:

```
            ['securedrop_app_https_certificate_cert_src', '', str,
             u'Local filepath to HTTPS certificate '
             '(optional, only if using HTTPS on source interface)',
             SiteConfig.ValidateOptionalPath(self.args.ansible_path),
             None,
             lambda config: config.get(
                 'securedrop_app_https_on_source_interface')],
```

This question will only be asked if `securedrop_app_https_on_source_interface` has been answered affirmatively.

## Testing

To test this PR, run `securedrop-admin sdconfig`. If you answer `yes` for the HTTPS question, you will be asked three followup questions regarding TLS certs. If you answer `no`, those questions will be skipped.

## Deployment

This does not impact existing deployments since it only concerns the setup flow.

## Checklist

N/A (none of the checklist items seem relevant to this change).

### CODE REVIEW

**Comment by dachary** [comment]
@antsar looks good :-) Would you be so kind as to add a test that verifies the implemented logic?

**Comment by antsar** [comment]
@dachary Thanks for the quick review! I've added a test.

**Comment by dachary** [comment]
> ./admin/tests/test_securedrop-admin.py:697:9: E303 too many blank lines (2)

The test looks great :+1:

File: admin/securedrop_admin/__init__.py [hunk 1]
```diff
@@ -255,115 +255,143 @@ def __init__(self, args):
...
+             u'Local filepath to HTTPS certificate ',
              SiteConfig.ValidateOptionalPath(self.args.ansible_path),
-             None],
+             None,
+             lambda config: config.get(
+                'securedrop_app_https_on_source_interface')],
             ['securedrop_app_https_certificate_key_src', '', str,
-             u'Local filepath to HTTPS certificate key '
-             '(optional, only if using HTTPS on source interface)',
+             u'Local filepath to HTTPS certificate key ',
```
[the remark this answers is not in the document]
Comment (antsar) [reply admin/securedrop_admin/__init__.py]: Thanks for catching this. Fixed.

**Pull request merged**